src/term/mod.rs

```diff
@@ -1352,6 +1353,22 @@ impl ansi::Handler for Term {
     fn set_title(&mut self, title: &str) {
         if self.dynamic_title {
             self.next_title = Some(title.to_owned());
+
+            #[cfg(windows)]
+            {
+                // cmd.exe in winpty: winpty incorrectly sets the title to ' ' instead of
+                // 'Alacritty' - thus we have to substitute this back to get equivalent
+                // behaviour as conpty.
+                //
+                // The startswith check is necessary because other shells e.g. bash send a
+                // different title and don't need Alacritty prepended.
+                if !tty::is_conpty() && title.starts_with(' ') {
+                    self.next_title = Some(match title.get(1..) {
+                        None => "Alacritty".into(),
+                        Some(rest) => format!("Alacritty{}", rest)
+                    });
```

Comment:
Is there any difference between `"Alacritty"` and `"Alacritty "` as a Window title? It seems like if `" "` is really used to reset it instead of `""`, then just using `format!("Alacritty{}", title)` would be fine.

It's also possible to just slap a `trim()` on the title at any point, which would remove leading and trailing whitespace, so even if it makes a difference I think that would be an easier option than having to match the first character.

Reply:
Thanks - I think avoiding the double space is desirable (it's noticable, just barely) - using `trim()` is a much neater solution though so I've switched to that.